openslides/motions/static/js/motions/site.js

```diff
@@ -560,9 +571,128 @@ angular.module('OpenSlidesApp.motions.site', ['OpenSlidesApp.motions'])
+        $scope.separator = ';';
+        $scope.encoding = 'UTF-8';
+        $scope.encodingOptions = ['UTF-8', 'ISO-8859-1'];
+        $scope.csv = {
+            content: null,
+            header: true,
+            headerVisible: false,
+            separator: $scope.separator,
+            separatorVisible: false,
+            encoding: $scope.encoding,
+            encodingVisible: false,
+            result: null
+        };
+        // set csv file encoding
+        $scope.setEncoding = function () {
+            $scope.csv.encoding = $scope.encoding;
+        };
+        // set csv file encoding
+        $scope.setSeparator = function () {
+            $scope.csv.separator = $scope.separator;
+        };
+        // detect if csv file is loaded
+        $scope.$watch('csv.result', function () {
+            angular.forEach($scope.csv.result, function (motion) {
+                if (motion.identifier)
+                    motion.identifier = motion.identifier.replace(/^"(.*)"$/, '$1');
+                if (motion.title)
+                    motion.title = motion.title.replace(/^"(.*)"$/, '$1');
+                if (motion.text)
+                    motion.text = motion.text.replace(/^"(.*)"$/, '$1');
+                if (motion.reason)
+                    motion.reason = motion.reason.replace(/^"(.*)"$/, '$1');
+                if (motion.submitter)
+                    motion.submitter = motion.submitter.replace(/^"(.*)"$/, '$1');
+                if (motion.category)
+                    motion.category = motion.category.replace(/^"(.*)"$/, '$1');
+            });
+        });
+
+        // import from csv file
+        $scope.importByCSV = function (result) {
+            var obj = JSON.parse(JSON.stringify(result));
+            $scope.csvimporting = true;
+            $scope.csvlines = Object.keys(obj).length;
+            $scope.csvimportcounter = 0;
+            for (var i = 0; i < obj.length; i++) {
+                var motion = {};
+                if (obj[i].identifier != '') {
+                    var motions = Motion.getAll();
+                    var exists = false;
+                    angular.forEach(motions, function (motion) {
+                        if (motion.identifier == obj[i].identifier) {
+                            exists = true;
+                        }
+                    });
+                    if (!exists) {
+                        motion.identifier = obj[i].identifier;
+                    }
+                    else {
+                        // abort current motion (iteration of loop);
+                        // TODO: show error
+                        continue;
+                    }
+                }
+                motion.title = obj[i].title;
+                motion.text = obj[i].text;
+                motion.reason = obj[i].reason;
+                motion.submitters_id = [];
+                if (obj[i].submitter != '') {
+                    var users = User.getAll();
```

Comment:
This does only get the users, that are already loaded by the client, but does not send a request to the server, to get all users. To get all users, you have to use User.fetchAll(), which will return a promise.